server/authenticator.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use system::concepts::Credential;
+use tonic::{Request, Status};
+use user::user_manager::UserManager;
+
+#[derive(Debug)]
+pub struct Authenticator {
+    user_manager: Arc<UserManager>,
+}
+
+impl Authenticator {
+    pub(crate) fn new(user_manager: Arc<UserManager>) -> Self {
+        Self { user_manager }
+    }
+}
+
+impl Authenticator {
+    pub fn authenticate(&self, req: Request<()>) -> Result<Request<()>, Status> {
+        let metadata = req.metadata();
+        let username_metadata = metadata.get("username").map(|u| u.to_str());
```

Comment:
These should be constants, too

Reply:
Fixed in https://github.com/typedb/typedb/commit/b9c8e80f8260a7b32a597e6374418e00cdf7d5d7